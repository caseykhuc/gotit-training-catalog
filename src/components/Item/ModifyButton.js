import React from 'react';
import { connect } from 'react-redux';
import { ButtonGroup, Button } from 'react-bootstrap';
import modalKeys from 'constants/modelKeys';
import { showModal } from 'actions/modal';
import PropTypes from 'prop-types';

export const ModifyButton = ({ categoryId, itemId, showModal }) => (
  <ButtonGroup>
    <Button className="mr-1">
      Edit
    </Button>
    <Button
      variant="danger"
      onClick={() => showModal(modalKeys.DELETE_ITEM_MODAL, {
        categoryId,
        itemId,
      })}
    >
      Delete
    </Button>
  </ButtonGroup>
)

ModifyButton.propTypes = {
  categoryId: PropTypes.number.isRequired,
  itemId: PropTypes.number.isRequired,
  showModal: PropTypes.func.isRequired,
}

export default connect(null, { showModal })(ModifyButton);
