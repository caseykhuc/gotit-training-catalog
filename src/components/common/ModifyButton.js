import React from 'react';
import { connect } from 'react-redux';
import { ButtonGroup, Button } from 'react-bootstrap';
import modalKeys from 'constants/modelKeys';
import { showModal } from 'actions/modal';
import PropTypes from 'prop-types';

export const ModifyButton = ({
  categoryId, itemId, currentValue, showModal, onDeleteSuccess,
}) => (
  <ButtonGroup>
    <Button
      className="mr-1"
      onClick={() => showModal(modalKeys.EDIT_ITEM_MODAL, {
        categoryId,
        itemId,
        currentValue,
      })}
    >
        Edit
    </Button>
    <Button
      variant="danger"
      onClick={() => showModal(modalKeys.DELETE_ITEM_MODAL, {
        categoryId,
        itemId,
        onSuccess: onDeleteSuccess,
      })}
    >
        Delete
    </Button>
  </ButtonGroup>
)

ModifyButton.propTypes = {
  categoryId: PropTypes.number.isRequired,
  itemId: PropTypes.number.isRequired,
  showModal: PropTypes.func.isRequired,
  currentValue: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    price: PropTypes.number,
  }),
  onDeleteSuccess: PropTypes.func,
}

export default connect(null, { showModal })(ModifyButton);
