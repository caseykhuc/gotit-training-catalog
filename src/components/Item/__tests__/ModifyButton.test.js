import React from 'react';
import { shallow } from 'enzyme';
import { ModifyButton } from '../ModifyButton';

describe('component/Iutem/ModifyButton', () => {
  let props;
  let wrapper;
  let deleteBtn;
  let editBtn;

  const update = () => {
    wrapper.update();
    deleteBtn = wrapper.find('[variant="danger"]');
    editBtn = wrapper.findWhere((x) => x.text() === 'Edit')
  };
  const setup = () => {
    wrapper = shallow(<ModifyButton {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      categoryId: 1,
      itemId: 10,
      showModal: jest.fn(),
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke showModal when Button is clicked', () => {
    setup();
    expect(deleteBtn).toHaveLength(1);
    deleteBtn.simulate('click', {});
    expect(props.showModal).toHaveBeenCalled();
    /* expect(editBtn).toHaveLength(1); */
  })
})
