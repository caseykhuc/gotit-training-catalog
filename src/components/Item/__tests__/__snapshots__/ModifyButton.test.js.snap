// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Item/ModifyButton should render correctly 1`] = `ShallowWrapper {}`;

exports[`component/Item/ModifyButton should render correctly 2`] = `ShallowWrapper {}`;

exports[`component/Iutem/ModifyButton should render correctly 1`] = `ShallowWrapper {}`;
